### PR TITLE
Update/ad-reporting-v2

### PR DESCRIPTION
**Pull request by fivetran-sheringuyen**
Pull Request
**Are you a current Fivetran customer?** 
Fivetran Employee 

**What change(s) does this PR introduce?** 

🚨 Breaking Changes 🚨
- `microsoft_ads__ad_adapter` report has been renamed to `microsoft_ads__url_report` to more accurately reflect contents of report.

🎉 Feature Enhancements 🎉
- Models have been updated to use level specific performance reporting for more accurate reporting.
- New models have been added:
  - `microsoft_ads__ad_report`
  - `microsoft_ads__keyword_report`
  - `microsoft_ads__search_report`
- New fields have been added to old models.
- `README` updates for easier navigation and use of the package.
- Passthrough metric functionality for more flexible reporting.
- Added testing for better data integrity.

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
😌 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
+# dbt_microsoft_ads v0.5.0
+## 🚨 Breaking Changes
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:2]: Can you add the PR link to the changelog

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-sheringuyen thanks so much for putting this PR together! Your updates look mostly good, I just had a few comments inline that should be addressed before we are good to go.
- [x] A few minor README and CHANGELOG updates.
- [ ] A question about including the `match_type` field to the keyword report
- [ ] Question around filtering the url report.
- [ ] Addressing any other inline comments I have from this review.

I also did take a closer look at the url report and the logic you applied makes sense to me! I really appreciated your addition to the DECISIONLOG about the auto tagging nature within Microsoft Ads.

File: models/microsoft_ads__keyword_report.sql [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        report.date_day,
+        accounts.account_name,
+        accounts.account_id,
+        campaigns.campaign_name,
+        campaigns.campaign_id,
+        ad_groups.ad_group_name,
+        ad_groups.ad_group_id,
+        ads.ad_name,
+        ads.ad_id,
+        keywords.keyword_name,
```
[the remark this answers is not in the document]
Comment (fivetran-sheringuyen) [reply models/microsoft_ads__keyword_report.sql:59]: Ah, I actually think I went into this really gnarly black hole regarding match types (Microsoft has several different kinds) and forgot to switch it to `*_history.match_type`. Good catch, thank you!

**Comment by fivetran-sheringuyen** [comment]
Hey @fivetran-joemarkiewicz, just updated the PR with your suggested fixes. Let me know if you have any other questions!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks for making the updates @fivetran-sheringuyen! These look good to go on my end 😄

**Pull request merged**